syntax = "proto3";

import "google/protobuf/empty.proto";

package mapreduce;


service Master {
    rpc RegisterService(RegisterServiceMes) returns (google.protobuf.Empty) {};
    rpc MapReduce(MapReduceRequest) returns (MapReduceResponse) {};
}

message RegisterServiceMes {
    string service_address = 1;
    int32 service_port = 2;
}

message MapReduceRequest {
    string input_dir = 1;
    int32 num_partitions = 2;
    optional string work_dir = 3;
    string mapper_path = 4;
    string reducer_path = 5;
}

message MapReduceResponse {
    string output_dir = 1;
}